sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ColumnPruningSuite.scala

```diff
@@ -249,5 +249,26 @@ class ColumnPruningSuite extends PlanTest {
     comparePlans(Optimize.execute(query), expected)
   }
 
+  test("Remove redundant Project introduced by column pruning rule") {
+    val input = LocalRelation('key.int, 'value.string)
+
+    val query =
+      Project(Seq($"x.key", $"y.key"),
```

Comment:
Hmm, maybe I shouldn't say that removing redundant Project introduced by column pruning. It is not accurate. The Project is becoming redundant after column pruning optimizes the plan.